recipes/xnnpack/all/conanfile.py

```diff
@@ -0,0 +1,101 @@
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+import glob
+import os
+
+
+class XnnpackConan(ConanFile):
+    name = "xnnpack"
+    description = "XNNPACK is a highly optimized library of floating-point " \
+                  "neural network inference operators for ARM, WebAssembly, " \
+                  "and x86 platforms."
+    license = "BSD-3-Clause"
+    topics = ("conan", "xnnpack", "neural-network", "inference", "multithreading",
+              "inference-optimization", "matrix-multiplication", "simd")
+    homepage = "https://github.com/google/XNNPACK"
+    url = "https://github.com/conan-io/conan-center-index"
+
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "assembly": [True, False],
+        "memopt": [True, False],
+        "sparse": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "assembly": True,
+        "memopt": True,
+        "sparse": True,
+    }
+
+    exports_sources = "CMakeLists.txt"
+    generators = "cmake"
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+        compiler = self.settings.compiler
+        compiler_version = tools.Version(compiler.version)
+        if (compiler == "gcc" and compiler_version < "6") or \
+           (compiler == "clang" and compiler_version < "5") or \
+           (compiler == "Visual Studio" and compiler_version < "16"):
+            raise ConanInvalidConfiguration("xnnpack doesn't support {} {}".format(str(compiler), compiler.version))
+
+    def requirements(self):
+        self.requires("cpuinfo/cci.20201217")
+        self.requires("fp16/cci.20200514")
+        self.requires("fxdiv/cci.20200417")
+        self.requires("pthreadpool/cci.20210218")
```

Comment:
Now I can see all pieces of puzzle together! Great!

Reply:
And if you want a first draft of what `requirements()` of `libtorch` will look like:
https://github.com/SpaceIm/conan-libtorch/blob/790b8e8ff2e64dcda70b7769e573ebfc40bbe1cf/conanfile.py#L153-L232

Reply:
![hide](https://media4.giphy.com/media/WRQiZ1rkueNh0Dw1mQ/200.gif)